### ISSUE TITLE
'list' object has no attribute 'min_occurs'

### ISSUE THREAD

**Issue by iLoveTux** [mvantellingen/python-zeep#44]
I am just about positive that this is not a problem with zeep, just thought I might reach out for help. Just a note, I am trying to wrap a web service which is defined by a WSDL which comes from a company which seems to be known (at least in my circles) for not following standards (or making up their own as they go).

So I need to ask first if it's supported to use a WSDL from the local filesystem instead of from a URL? I didn't see it mentioned in the documentation, but figured I'd give it a try anyway. My code looks like this:

```python
import zeep

client = zeep.Client("./proprietary.wsdl")
```

At this point I get this traceback:

```
Traceback (most recent call last):
  File "t.py", line 3, in <module>
    client = zeep.Client('proprietary.wsdl')
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/client.py", line 45, in __init__
    self.wsdl = Document(wsdl, self.transport)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/wsdl/wsdl.py", line 58, in __init__
    root_definitions = Definition(self, document, self.location)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/wsdl/wsdl.py", line 147, in __init__
    self.schema = self.parse_types(doc)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/wsdl/wsdl.py", line 267, in parse_types
    self.wsdl._parser_context, self.location)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/schema.py", line 47, in __init__
    visitor.visit_schema(node)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 94, in visit_schema
    self.process(node, parent=parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 41, in process
    result = visit_func(self, node, parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 157, in visit_include
    return self.visit_schema(schema_node)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 94, in visit_schema
    self.process(node, parent=parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 41, in process
    result = visit_func(self, node, parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 209, in visit_element
    xsd_type = self.process(child, node)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 41, in process
    result = visit_func(self, node, parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 374, in visit_complex_type
    item = self.process(child, node)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 41, in process
    result = visit_func(self, node, parent)
  File "/home/ilovetux/anaconda3/lib/python3.5/site-packages/zeep/xsd/visitor.py", line 733, in visit_choice
    elm.min_occurs = 0
AttributeError: 'list' object has no attribute 'min_occurs'
```

Honestly, I was expecting to get a connection error or something similar, for which I might simply offer a pull-request, but this seems like it's a problem with the actual WSDL.

Trying to debug the situation, I changed [this method](https://github.com/mvantellingen/python-zeep/blob/master/src/zeep/xsd/visitor.py#L715) to read as follows:

```python
    def visit_choice(self, node, parent):
        """
            <choice
              id = ID
              maxOccurs= (nonNegativeInteger | unbounded) : 1
              minOccurs= nonNegativeInteger : 1
              {any attributes with non-schema Namespace}...>
            Content: (annotation?, (element | group | choice | sequence | any)*)
            </choice>
        """
        # There should be only max nodes, first node (annotation) is irrelevant
        children = node.getchildren()
        if children[0].tag == tags.annotation:
            children.pop(0)

        choices = []
        for child in children:
            elm = self.process(child, node)
            try:
                elm.min_occurs = 0
            except:
                print(etree.tostring(child, pretty_print=True).decode("utf-8"))
                raise
            choices.append(elm)
        return xsd_elements.Choice(choices)
```

And in addition to the above traceback I got this (I'm not sure about license, so I took out anything identifying):

```xml
<xsd:sequence xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns1="http://www.proprietary.com/URI">
    <xsd:element name="UniqueElement-1" type="xsd:string"/>
    <xsd:element name="UniqueElement-2" type="xsd:string"/>
    <xsd:element name="UniqueElement-3" type="ns1:CustomType-1"/>
    <xsd:element name="UniqueElement-4" type="ns1:CustomType-2"/>
    <xsd:element name="UniqueElement-5" type="xsd:string"/>
    <xsd:element name="UniqueElement-6" type="xsd:boolean"/>
    <xsd:element name="UniqueElement-7" type="xsd:string" minOccurs="0" maxOccurs="1"/>
    <xsd:element name="UniqueElement-8" type="ns1:CustomType-3" minOccurs="0" maxOccurs="unbounded"/>
    <xsd:element name="UniqueElement-9" type="ns1:CustomType-4" minOccurs="0" maxOccurs="1"/>
    <xsd:element name="UniqueElement-10" type="ns1:CustomType-5" minOccurs="0" maxOccurs="unbounded"/>
    <xsd:element name="UniqueElement-11" type="ns1:CustomType-6" minOccurs="0" maxOccurs="unbounded"/>
</xsd:sequence>
```

To my untrained eye, it looks ok, but it's too late tonight for me to go pouring over the WSDL/XSD spec. Is the problem amazingly obvious and I'm just missing it?

**Comment by mvantellingen** [comment]
Thanks for the detailed report! So the first thing to check is if SOAPUI can load the WSDL. If the wsdl works in SOAPUI and not in Zeep then I consider it a bug :-)

See https://www.soapui.org/downloads/soapui.html

**Comment by iLoveTux** [comment]
SoapUI can successfully load the WSDL, I use SoapUI extensively with this web service, I was just looking into the possibility of scripting interactions with Python, before I saw your project, I was going to handle the XML with string manipulation.

**Comment by mvantellingen** [comment]
Oke, this is something i'd like to fix. Are you able to make the complextype with the choice public?

**Comment by iLoveTux** [comment]
I've had to sanitize it again, but here it is

```xml
<xsd:element name='ElementName'>
    <xsd:complexType xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns1="http://www.proprietary.com/URI">
      <xsd:choice>
        <xsd:sequence>
            <xsd:element name="UniqueElement-1" type="xsd:string"/>
            <xsd:element name="UniqueElement-2" type="xsd:string"/>
            <xsd:element name="UniqueElement-3" type="ns1:CustomType-1"/>
            <xsd:element name="UniqueElement-4" type="ns1:CustomType-2"/>
            <xsd:element name="UniqueElement-5" type="xsd:string"/>
            <xsd:element name="UniqueElement-5" type="xsd:boolean"/>
            <xsd:element name="UniqueElement-6" type="xsd:string" minOccurs="0" maxOccurs="1"/>
            <xsd:element name="UniqueElement-7" type="ns1:CustomType-3" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="UniqueElement-8" type="ns1:CustomType-4" minOccurs="0" maxOccurs="1"/>
            <xsd:element name="UniqueElement-9" type="ns1:CustomType-5" minOccurs="0" maxOccurs="unbounded"/>
            <xsd:element name="UniqueElement-10" type="ns1:CustomType-6" minOccurs="0" maxOccurs="unbounded"/>
        </xsd:sequence>
      </xsd:choice>
    </xsd:complexType>
</xsd:element>
```

**Comment by mvantellingen** [comment]
Thanks! 👍

**Comment by mvantellingen** [comment]
Should be fixed with #55

**Comment by iLoveTux** [comment]
I apologize for not getting back to you, its been hectic around here. I can verify that this indeed fixes the issue.

There is, however, an issue parsing the response, but I will open a new issue for that.

Thank you by the way for all the work you are putting into this, it will be incredibly useful for me and it seems like a lot of other people as well. Is there any way I can show my appreciation monetarily, I'm not rich but I do like to show my appreciation for incredibly useful open source projects.
